### PR TITLE
Better diagnostics error when a user tries to format code that has parsing diagnostics.

### PR DESCRIPTION
**Pull request by shoaibkh4n**
This pull request addresses the issue outlined in [#1698](https://github.com/biomejs/biome/issues/1698#issue-2105338537), aiming to enhance the diagnostic message returned to the user when attempting to format code that contains parsing errors. 

Previously, the message was not very informative or actionable, potentially leaving users confused about why their code formatting failed and how they could resolve the issue.

Changes Made:
Updated the FormatWithErrorsDisabled diagnostic message in [biome/crates/biome_service/src/diagnostics.rs](https://github.com/biomejs/biome/blob/13ec78dc3bc4bded1bca8072f9a55a6756f32cf8/crates/biome_service/src/diagnostics.rs#L391-L393) to clearly explain why the formatting operation was aborted and provide users with guidance on how to enable formatting in the presence of parsing errors.

New Diagnostic Message:

```
#[derive(Debug, Serialize, Deserialize, Diagnostic)]
#[diagnostic(
    category = "format",
    message = "Code formatting aborted due to parsing errors. To format code with errors, enable the 'formatter.formatWithError' option."
)]
pub struct FormatWithErrorsDisabled;
```

This pull request seeks to merge these changes into the main branch of the biome project, improving the clarity and usefulness of diagnostics related to formatting code with errors.

### CODE REVIEW

File: crates/biome_service/src/diagnostics.rs [hunk 1]
```diff
@@ -389,7 +389,7 @@ pub struct ReportNotSerializable {
 pub struct NotFound;
 
 #[derive(Debug, Serialize, Deserialize, Diagnostic)]
-#[diagnostic(category = "format", message = "Format with errors is disabled.")]
+#[diagnostic(category = "format", message = "Code formatting aborted due to parsing errors. To format code with errors, enable the 'formatter.formatWithError' option.")]
```
Comment (ematipico) [inline crates/biome_service/src/diagnostics.rs:392]:
```suggestion
#[diagnostic(category = "format", message = "Code formatting aborted due to parsing errors. To format code with errors, enable the 'formatter.formatWithErrors' option.")]
```

**Review by ematipico — APPROVED** [review]
Awesome! Thank you

**Comment by shoaibkh4n** [comment]
Hey @ematipico ,
I think some checks got failed, Could you please verify them "Format and Lint Rust Files (pull_request)" and "
Pull request / Test (ubuntu-latest) (pull_request)".

**Comment by unvalley** [comment]
@shoaibkh4n Executing `cargo fmt` should resolve the Format and Lint Rust Files error.

**Comment by ematipico** [comment]
> Hey @ematipico , I think some checks got failed, Could you please verify them "Format and Lint Rust Files (pull_request)" and " Pull request / Test (ubuntu-latest) (pull_request)".

Go inside the folder `crates/biome_cli`, run `cargo test` and then `cargo insta accept`. Commit the new changes. 
Also, apply @unvalley suggestion.

**Comment by shoaibkh4n** [comment]
@unvalley @ematipico , Made the changes successfully and it might gonna work this time. I think it requires approval one more time, could you please verify and approve it. Thanks

**Pull request merged**